src/libertem/io/dataset/k2is.py

```diff
@@ -541,17 +547,27 @@ def __init__(self, path, io_backend=None):
         super().__init__(io_backend=io_backend)
         self._path = path
         self._start_offsets = None
-        # NOTE: the sync flag appears to be set one frame too late, so
-        # we compensate here by setting a negative _skip_frames value.
-        # skip_frames is applied after synchronization.
-        self._skip_frames = -1
         self._files = None
         self._sync_offset = 0
 
     def _do_initialize(self):
         self._files = self._get_files()
+        with dm.fileDM(_get_gtg_path(self._path), on_memory=True) as dm_file:
+            nav_shape_y = dm_file.allTags.get('.SI Dimensions.Size Y')
+            nav_shape_x = dm_file.allTags.get('.SI Dimensions.Size X')
```

Comment:
Is there a reason this can't stay in its own method?

Reply:
Ah, right. I could use a setter.